### PR TITLE
Small fixes and small refactorings toward reuse in CWL branch

### PR DESCRIPTION
**Pull request by jmchilton**
These are really simple things so I'm aggregating them but I can explode them out into individual PRs (all at once or for individual commits if anything warrants additional discussion). These are from the CWL fork of Galaxy.

### CODE REVIEW

**Review by mvdbeek — APPROVED** [review]

File: lib/galaxy/tools/__init__.py [hunk 1]
```diff
@@ -386,8 +386,13 @@ class Tool(object, Dictifiable):
...
         """Load a tool from the config named by `config_file`"""
         # Determine the full path of the directory where the tool config is
-        self.config_file = config_file
-        self.tool_dir = os.path.dirname(config_file)
+        if config_file is not None:
+            self.config_file = config_file
+            self.tool_dir = os.path.dirname(config_file)
+        else:
+            self.config_file = None
+            self.tool_dir = None
```
Comment (nsoranzo) [inline lib/galaxy/tools/__init__.py:394]: What about:
```python
        self.config_file = config_file
        if self.config_file is not None:
            self.tool_dir = os.path.dirname(self.config_file)
        else:
            self.tool_dir = None
```

File: test/base/populators.py [hunk 2]
```diff
@@ -246,9 +246,18 @@ def get_remote_files(self, target="ftp"):
         return self._get("remote_files", data={"target": target}).json()
 
     def run_tool_payload(self, tool_id, inputs, history_id, **kwds):
-        if "files_0|file_data" in inputs:
-            kwds["__files"] = {"files_0|file_data": inputs["files_0|file_data"]}
-            del inputs["files_0|file_data"]
+        # Remove files_%d|file_data parameters from inputs dict and attach
+        # as __files dictionary.
+        file_keys = set()
```
Comment (nsoranzo) [inline test/base/populators.py:251]: You don't need `file_keys` if you create a copy of `inputs`, e.g.:
```python
        for key, value in list(inputs.items()):
            if key.startswith("files_") and key.endswith("|file_data"):
                if "__files" not in kwds:
                    kwds["__files"] = {}
                kwds["__files"][key] = value
                del inputs[key]
```

**Comment by nsoranzo** [comment]
There are a few other places where the tool directory is assumed to be not None:
- https://github.com/jmchilton/galaxy/blob/3f8a49fac669279149cc9b3ce4ad06c50979bb5b/lib/galaxy/jobs/__init__.py#L877
- https://github.com/jmchilton/galaxy/blob/3f8a49fac669279149cc9b3ce4ad06c50979bb5b/lib/galaxy/managers/citations.py#L29
- https://github.com/jmchilton/galaxy/blob/3f8a49fac669279149cc9b3ce4ad06c50979bb5b/lib/galaxy/tools/deps/resolvers/brewed_tool_shed_packages.py#L46
- https://github.com/jmchilton/galaxy/blob/4c7a1cfc767ffa666b053c526072b5f5c5f06c6c/lib/galaxy/tools/evaluation.py#L483
- https://github.com/jmchilton/galaxy/blob/3f8a49fac669279149cc9b3ce4ad06c50979bb5b/lib/galaxy/tools/__init__.py#L655
- https://github.com/jmchilton/galaxy/blob/3f8a49fac669279149cc9b3ce4ad06c50979bb5b/lib/galaxy/tools/__init__.py#L752

I'm not sure if checks need to be added there.

**Comment by jmchilton** [comment]
> There are a few other places where the tool directory is assumed to be not None:

I hadn't hit these yet downstream with file-less tools but I was going to eventually probably so I'll go through and look at these - thanks.

I'll also incorporate these other changes - thanks for the detailed review @nsoranzo!

**Pull request closed without merging**